libbeat/generator/fields/fields.go

```diff
@@ -39,37 +32,67 @@ type YmlFile struct {
 	Indent int
 }
 
-func collectBeatFiles(beatPath string, fieldFiles []*YmlFile) ([]*YmlFile, error) {
-	commonFields := filepath.Join(beatPath, "_meta", "fields.common.yml")
-	_, err := os.Stat(commonFields)
-	if os.IsNotExist(err) {
-		return fieldFiles, nil
-	} else if err != nil {
-		return nil, err
+func collectCommonFiles(esBeatsPath, beatPath string, fieldFiles []*YmlFile) ([]*YmlFile, error) {
+	commonFields := []string{
+		filepath.Join(beatPath, "_meta", "fields.common.yml"),
+	}
+
+	var libbeatFieldFiles []*YmlFile
+	var err error
+	if !isLibbeat(beatPath) {
+		commonFields = append(commonFields,
+			filepath.Join(esBeatsPath, "libbeat", "_meta", "fields.common.yml"),
```

Comment:
BTW `filepath.Join` always calls `filepath.Clean` on the result. https://golang.org/pkg/path/filepath/#Join

This means you can write `filepath.Join(esBeatsPath, "libbeat/_meta/fields.common.yml")` which is more clear IMO.